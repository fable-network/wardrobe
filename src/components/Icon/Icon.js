import React from 'react';
import PropTypes from 'prop-types';
import '../../static/iconsprite.svg';

console.log(window.ICONS_LIST);
/**
 * Uses a SVG sprite to show icons based on name
 */
const Icon = ({ name, color, fill, stroke, width, height, strokeWidth, ...otherProps }) => (
  <svg
    className="ft-icon"
    width={width}
    height={height}
    fill={fill || color}
    stroke={stroke}
    strokeWidth={strokeWidth}
    {...otherProps}
  >
    <use xlinkHref={`/iconsprite.svg#icon-${name}`} />
  </svg>
);

Icon.propTypes = {
  /** Name of the icon as defined in the sprite */
  name: PropTypes.string.isRequired,

  /**
   * Color for the SVG `fill`  attributes
   * @deprecated
   * */
  color: PropTypes.string,
  fill: PropTypes.string,
  stroke: PropTypes.string,
  /** Change width; SVG ratio will be preserved */
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** Change height; SVG ratio will be preserved */
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  strokeWidth: PropTypes.number,
};

Icon.defaultProps = {
  color: 'inherit',
  width: '1em',
  height: '1em',
  strokeWidth: 1,
};

export default Icon;
