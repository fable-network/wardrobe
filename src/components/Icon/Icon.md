**Default icon display**

```jsx
<Icon name="caret-down" />
```

**Custom color (orange) set on the Icon itself**

```jsx
<Icon name="caret-down" fill="orange" />
```

**Custom color (green) set by the `fill` attribute on a parent element**

```jsx
<div style={{ fill: 'green' }}>
  <Icon name="caret-down" />
</div>
```

**width / height**

When using a custom size, the aspect ratio of the icon is preserved.
Whitespace will be added to create the desired width and height.

Custom size (100x20px), background color set to `khaki` to outline the SVG:

```jsx
<Icon name="caret-down" width="100" height="20" style={{ background: 'khaki', fill: 'navy' }} />
```

```jsx
<Icon name="caret-down" width="20" height="50" style={{ background: 'khaki', fill: 'navy' }} />
```

**All icons**

```jsx
<Grid>
  {iconsList.map(icon => (
    <Grid.Cell columns={3} key={icon}>
      <div style={{ marginBottom: '16px', display: 'flex', justifyContent: 'center' }}>
        <Stack>
          <div style={{ margin: '16px', background: '#f0f0f0' }}>
            <Icon name={icon} width={32} height={32} />
          </div>
          <div>{icon}</div>
        </Stack>
      </div>
    </Grid.Cell>
  ))}
</Grid>
```
