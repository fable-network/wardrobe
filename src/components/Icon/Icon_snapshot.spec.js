/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import Icon from './Icon';

describe('Icon Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = shallow(<Icon name="caret-down" />);
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = shallow(<Icon name="caret-down" color="orange" />);
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = shallow(<Icon name="caret-down" width="100" height="20" style={{background: 'khaki', fill: 'navy'}}/>);
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = shallow(<Icon name="caret-down" width="20" height="50" style={{background: 'khaki', fill: 'navy'}}/>);
      expect(wrapper4).toMatchSnapshot();

    });
  });
});
