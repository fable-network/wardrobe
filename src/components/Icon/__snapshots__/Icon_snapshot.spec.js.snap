// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    color="inherit"
    height="1em"
    name="caret-down"
    strokeWidth={1}
    width="1em"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <use
        xlinkHref="/iconsprite.svg#icon-caret-down"
      />,
      "className": "ft-icon",
      "fill": "inherit",
      "height": "1em",
      "stroke": undefined,
      "strokeWidth": 1,
      "width": "1em",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "xlinkHref": "/iconsprite.svg#icon-caret-down",
      },
      "ref": null,
      "rendered": null,
      "type": "use",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Icon__Svg-sf8tqo-0",
        "isStatic": false,
        "rules": Array [
          "width:",
          [Function],
          ";height ",
          [Function],
          ";",
        ],
      },
      "displayName": "Icon__Svg",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Icon__Svg-sf8tqo-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <use
          xlinkHref="/iconsprite.svg#icon-caret-down"
        />,
        "className": "ft-icon",
        "fill": "inherit",
        "height": "1em",
        "stroke": undefined,
        "strokeWidth": 1,
        "width": "1em",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "xlinkHref": "/iconsprite.svg#icon-caret-down",
        },
        "ref": null,
        "rendered": null,
        "type": "use",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Icon__Svg-sf8tqo-0",
          "isStatic": false,
          "rules": Array [
            "width:",
            [Function],
            ";height ",
            [Function],
            ";",
          ],
        },
        "displayName": "Icon__Svg",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Icon__Svg-sf8tqo-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Icon Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    color="inherit"
    fill="orange"
    height="1em"
    name="caret-down"
    strokeWidth={1}
    width="1em"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <use
        xlinkHref="/iconsprite.svg#icon-caret-down"
      />,
      "className": "ft-icon",
      "fill": "orange",
      "height": "1em",
      "stroke": undefined,
      "strokeWidth": 1,
      "width": "1em",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "xlinkHref": "/iconsprite.svg#icon-caret-down",
      },
      "ref": null,
      "rendered": null,
      "type": "use",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Icon__Svg-sf8tqo-0",
        "isStatic": false,
        "rules": Array [
          "width:",
          [Function],
          ";height ",
          [Function],
          ";",
        ],
      },
      "displayName": "Icon__Svg",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Icon__Svg-sf8tqo-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <use
          xlinkHref="/iconsprite.svg#icon-caret-down"
        />,
        "className": "ft-icon",
        "fill": "orange",
        "height": "1em",
        "stroke": undefined,
        "strokeWidth": 1,
        "width": "1em",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "xlinkHref": "/iconsprite.svg#icon-caret-down",
        },
        "ref": null,
        "rendered": null,
        "type": "use",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Icon__Svg-sf8tqo-0",
          "isStatic": false,
          "rules": Array [
            "width:",
            [Function],
            ";height ",
            [Function],
            ";",
          ],
        },
        "displayName": "Icon__Svg",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Icon__Svg-sf8tqo-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Icon Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    color="inherit"
    height="20"
    name="caret-down"
    strokeWidth={1}
    style={
      Object {
        "background": "khaki",
        "fill": "navy",
      }
    }
    width="100"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <use
        xlinkHref="/iconsprite.svg#icon-caret-down"
      />,
      "className": "ft-icon",
      "fill": "inherit",
      "height": "20",
      "stroke": undefined,
      "strokeWidth": 1,
      "style": Object {
        "background": "khaki",
        "fill": "navy",
      },
      "width": "100",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "xlinkHref": "/iconsprite.svg#icon-caret-down",
      },
      "ref": null,
      "rendered": null,
      "type": "use",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Icon__Svg-sf8tqo-0",
        "isStatic": false,
        "rules": Array [
          "width:",
          [Function],
          ";height ",
          [Function],
          ";",
        ],
      },
      "displayName": "Icon__Svg",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Icon__Svg-sf8tqo-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <use
          xlinkHref="/iconsprite.svg#icon-caret-down"
        />,
        "className": "ft-icon",
        "fill": "inherit",
        "height": "20",
        "stroke": undefined,
        "strokeWidth": 1,
        "style": Object {
          "background": "khaki",
          "fill": "navy",
        },
        "width": "100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "xlinkHref": "/iconsprite.svg#icon-caret-down",
        },
        "ref": null,
        "rendered": null,
        "type": "use",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Icon__Svg-sf8tqo-0",
          "isStatic": false,
          "rules": Array [
            "width:",
            [Function],
            ";height ",
            [Function],
            ";",
          ],
        },
        "displayName": "Icon__Svg",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Icon__Svg-sf8tqo-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Icon Component snapshots should match snapshot 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    color="inherit"
    height="50"
    name="caret-down"
    strokeWidth={1}
    style={
      Object {
        "background": "khaki",
        "fill": "navy",
      }
    }
    width="20"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <use
        xlinkHref="/iconsprite.svg#icon-caret-down"
      />,
      "className": "ft-icon",
      "fill": "inherit",
      "height": "50",
      "stroke": undefined,
      "strokeWidth": 1,
      "style": Object {
        "background": "khaki",
        "fill": "navy",
      },
      "width": "20",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "xlinkHref": "/iconsprite.svg#icon-caret-down",
      },
      "ref": null,
      "rendered": null,
      "type": "use",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Icon__Svg-sf8tqo-0",
        "isStatic": false,
        "rules": Array [
          "width:",
          [Function],
          ";height ",
          [Function],
          ";",
        ],
      },
      "displayName": "Icon__Svg",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Icon__Svg-sf8tqo-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <use
          xlinkHref="/iconsprite.svg#icon-caret-down"
        />,
        "className": "ft-icon",
        "fill": "inherit",
        "height": "50",
        "stroke": undefined,
        "strokeWidth": 1,
        "style": Object {
          "background": "khaki",
          "fill": "navy",
        },
        "width": "20",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "xlinkHref": "/iconsprite.svg#icon-caret-down",
        },
        "ref": null,
        "rendered": null,
        "type": "use",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Icon__Svg-sf8tqo-0",
          "isStatic": false,
          "rules": Array [
            "width:",
            [Function],
            ";height ",
            [Function],
            ";",
          ],
        },
        "displayName": "Icon__Svg",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Icon__Svg-sf8tqo-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
